components/global_search/global-search.react.js

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import './style.scss';
+import SearchOption from "./child_components/search_option/SearchOption";
+
+class GlobalSearch extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.handle_search_selection = this.handle_search_selection.bind(this);
+        this.toggle_search_options = this.toggle_search_options.bind(this);
+        this.search_bar_focused = this.search_bar_focused.bind(this);
+        this.get_select_search_type = this.get_select_search_type.bind(this);
+        this.show_hide_mobile = this.show_hide_mobile.bind(this);
+
+        this.show_search_options_ref = React.createRef();
+        this.show_hide_mobile_ref = React.createRef();
+
+        this.state = {
```

Comment:
You could possibly remove this `this` keyword. It will still work.

Reply:
Me and @punalsc sat down and had issues implementing this, so we will leave it as is for now.